src/nvim/ui.c

```diff
@@ -320,11 +340,17 @@ void ui_set_ext_option(UI *ui, UIExtension ext, bool active)
 void ui_line(ScreenGrid *grid, int row, int startcol, int endcol, int clearcol,
              int clearattr, bool wrap)
 {
+  LineFlags flags = wrap ? kLineFlagWrap : 0;
+  if (startcol == -1) {
+    startcol = 0;
+    flags |= kLineFlagInvalid;
+  }
+
   size_t off = grid->line_offset[row] + (size_t)startcol;
 
-  UI_CALL(raw_line, grid->handle, row, startcol, endcol, clearcol, clearattr,
-          wrap, (const schar_T *)grid->chars + off,
-          (const sattr_T *)grid->attrs + off);
+  UI_CALL_IF(!ui->composed, raw_line, grid->handle, row, startcol, endcol,
+             clearcol, clearattr, flags, (const schar_T *)grid->chars + off,
+             (const sattr_T *)grid->attrs + off);
```

Comment:
Wouldn't it be clearer if this (and the other `UI_CALL_IF` cases) was a plain old if-statement? 

```c
if (!ui->composed) {
  UI_CALL(...);
}
```

Maybe `UI_CALL_IF` is not needed.

Reply:
@justinmk But all UI calls are already if statements in for-loop, this is nothing new with compositor... Just that the condition normally just is `if (ui->method) ui->method(...)` instead of `if (ui->method && cond)` (but for no reason two if-statements, I should merge them).

Also this change is from #9530, not this PR.